fd9cfc4db512f41612727a4a4fe0303fe8f67bef mentions that other packages were broken as well, but doesn't say which ones.  #41616 , which that fix is attached to, is about the `nixos.tests.gnome3` test.

I cannot build the gnome3 NixOS test with this PR.  I get this error while trying to build `gnome3.zenity`, one of its dependencies:
```
Traceback (most recent call last):
  File "/nix/store/0ri31bhv3n57pxqaky3s8sqz72yqay7s-itstool-2.0.6/bin/itstool", line 27, in <module>
    import libxml2
  File "/nix/store/gf8s6kl3v1fgqaz52qd2ffnvja9s8rjv-libxml2+py-2.9.9/lib/python2.7/site-packages/libxml2.py", line 1, in <module>
    import libxml2mod
ImportError: dynamic module does not define module export function (PyInit_libxml2mod)
make[2]: *** [Makefile:554: bg/bg.stamp] Error 1
make[2]: Leaving directory '/build/zenity-3.32.0/help'
make[1]: *** [Makefile:424: all-recursive] Error 1
make[1]: Leaving directory '/build/zenity-3.32.0'
make: *** [Makefile:365: all] Error 2
builder for '/nix/store/nixp439kx1acw8s0nxbbssky118fhc8b-zenity-3.32.0.drv' failed with exit code 2
error: build of '/nix/store/nixp439kx1acw8s0nxbbssky118fhc8b-zenity-3.32.0.drv' failed
```

I _can_ build zenity with the itstool-back-to-python2 PR #72335

--- a/pkgs/development/libraries/libxml2/default.nix
+++ b/pkgs/development/libraries/libxml2/default.nix
@@ -14,6 +14,20 @@ stdenv.mkDerivation rec {
     url = "http://xmlsoft.org/sources/${pname}-${version}.tar.gz";
     sha256 = "0wd881jzvqayx0ihzba29jl80k06xj9ywp16kxacdqs3064p1ywl";
   };
+  patches = [
+    # Upstream bugs:
+    #   https://bugzilla.gnome.org/show_bug.cgi?id=789714
+    #   https://gitlab.gnome.org/GNOME/libxml2/issues/64
+    # Patch from https://bugzilla.opensuse.org/show_bug.cgi?id=1065270 ,
+    # but only the UTF-8 part.
+    # Can also be mitigated by fixing malformed XML inputs, such as in
+    # https://gitlab.gnome.org/GNOME/gnumeric/merge_requests/3 .
+    # Other discussion:
+    #   https://github.com/itstool/itstool/issues/22
+    #   https://github.com/NixOS/nixpkgs/pull/63174
+    #   https://github.com/NixOS/nixpkgs/pull/72342
+    ./utf8-xmlErrorFuncHandler.patch
+  ];
 
   outputs = [ "bin" "dev" "out" "man" "doc" ]
     ++ lib.optional pythonSupport "py"

--- a/pkgs/development/libraries/libxml2/utf8-xmlErrorFuncHandler.patch
+++ b/pkgs/development/libraries/libxml2/utf8-xmlErrorFuncHandler.patch
@@ -1,0 +1,30 @@
+Index: libxml2-2.9.5/python/libxml.c
+===================================================================
+--- libxml2-2.9.5.orig/python/libxml.c
++++ libxml2-2.9.5/python/libxml.c
+@@ -1620,6 +1620,7 @@ libxml_xmlErrorFuncHandler(ATTRIBUTE_UNU
+     PyObject *message;
+     PyObject *result;
+     char str[1000];
++    unsigned char *ptr = (unsigned char *)str;
+ 
+ #ifdef DEBUG_ERROR
+     printf("libxml_xmlErrorFuncHandler(%p, %s, ...) called\n", ctx, msg);
+@@ -1636,10 +1637,16 @@ libxml_xmlErrorFuncHandler(ATTRIBUTE_UNU
+ 	    str[999] = 0;
+         va_end(ap);
+ 
++#if PY_MAJOR_VERSION >= 3
++        /* Ensure the error string doesn't start at UTF8 continuation. */
++        while (*ptr && (*ptr & 0xc0) == 0x80)
++            ptr++;
++#endif
++
+         list = PyTuple_New(2);
+         PyTuple_SetItem(list, 0, libxml_xmlPythonErrorFuncCtxt);
+         Py_XINCREF(libxml_xmlPythonErrorFuncCtxt);
+-        message = libxml_charPtrConstWrap(str);
++        message = libxml_charPtrConstWrap(ptr);
+         PyTuple_SetItem(list, 1, message);
+         result = PyEval_CallObject(libxml_xmlPythonErrorFuncHandler, list);
+         Py_XDECREF(list);

